azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/MainActivity.kt

```diff
@@ -21,58 +21,54 @@ import androidx.activity.result.contract.ActivityResultContracts
-import com.azure.android.communication.ui.redux.Store
 import com.azure.android.communication.ui.redux.action.CallingAction
-import com.azure.android.communication.ui.redux.middleware.handler.CallingMiddlewareActionHandler
 import com.azure.android.communication.ui.redux.state.NavigationStatus
-import com.azure.android.communication.ui.redux.state.ReduxState
 import com.azure.android.communication.ui.service.calling.InCallService
 import kotlinx.coroutines.launch
 
 internal class MainActivity : AppCompatActivity() {
 
     private val diContainerHolder: DependencyInjectionContainerHolder by viewModels()
-
-    private lateinit var navigationRouter: NavigationRouter
-    private lateinit var fragmentFactory: FragmentFactory
-    private lateinit var store: Store<ReduxState>
-    private lateinit var configuration: CallCompositeConfiguration
-    private lateinit var permissionManager: PermissionManager
-    private lateinit var audioSessionManager: AudioSessionManager
-    private lateinit var lifecycleManager: LifecycleManager
-    private lateinit var errorHandler: ErrorHandler
-    private lateinit var callingMiddlewareActionHandler: CallingMiddlewareActionHandler
-    private lateinit var videoViewManager: VideoViewManager
+    private val container by lazy { diContainerHolder.container }
+
+    private val navigationRouter get() = container.navigationRouter
+    private val fragmentFactory get() = container.fragmentFactory
+    private val store get() = container.appStore
+    private val configuration get() = container.configuration
+    private val permissionManager get() = container.permissionManager
+    private val audioSessionManager get() = container.audioSessionManager
+    private val lifecycleManager get() = container.lifecycleManager
+    private val errorHandler get() = container.errorHandler
+    private val callingMiddlewareActionHandler get() = container.callingMiddlewareActionHandler
+    private val videoViewManager get() = container.videoViewManager
+    private val instanceId get() = intent.getIntExtra(KEY_INSTANCE_ID, 0)
 
     override fun onDestroy() {
-        super.onDestroy()
         navigationRouter.removeOnNavigationStateChanged(this::onNavigationStateChange)
-
         if (isFinishing) {
-            DIContainer = null
             store.dispatch(CallingAction.CallEndRequested())
+
+            // /Clear this config
+            CallCompositeConfiguration.putConfig(instanceId, null)
         }
+        super.onDestroy()
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        injectDependencies()
+        super.onCreate(savedInstanceState)
+
+        // / Assign the DI Container the appropriate instanceId, so it can initialize it's container gi
+        diContainerHolder.instanceId = Integer.valueOf(instanceId)
         lifecycleScope.launch { errorHandler.start() }
         supportFragmentManager.fragmentFactory = fragmentFactory
```

Comment:
@iaulakh 
The problem you reported is here.

Since the fragmentFactory is inside the container, and the container is VM initialized (available after onCreate) and fragmentFactory is required to be set before onCreate() the fragment factory fails.

This might be a follow up issue. I'll investigate if there is a path to removing custom fragmentFactory (e.g. ViewModelProvider for sharing the activity view model and passing services to the child fragments) or another option, e.g. rescoping the FragmentFactory and some of it's dependencies so they are available outside ViewModel.

Reply:
Thanks :)